jobs/scrap_archives.py

```diff
@@ -0,0 +1,190 @@
+import asyncio
+import concurrent.futures
+import json
+import logging
+import re
+import sys
+from packaging.version import parse as version_parse
+
+import aiohttp
+import kinto_http.exceptions
+from kinto_http import cli_utils
+
+
+ARCHIVE_URL = "https://archive.mozilla.org/pub/"
+PRODUCTS = ("firefox", "thunderbird")
+DEFAULT_SERVER = "https://kinto-ota.dev.mozaws.net/v1"
+DEFAULT_BUCKET = "build-hub"
+DEFAULT_COLLECTION = "archives"
+NB_THREADS = 10
+
+
+logger = logging.getLogger(__name__)
+
+
+def publish_record(client, record):
+    try:
+        r = client.create_record(record)
+        logger.info("Created record %s" % json.dumps(r))
+    except kinto_http.exceptions.KintoException:
+        error_msg = "Could not create record for %s" % record["download"]["url"]
+        logger.exception(error_msg)
+
+
+def archive_url(product, version=None, platform=None, locale=None):
+    url = ARCHIVE_URL + product + "/releases/"
+    if version:
+        url += version + "/"
+    if platform:
+        url += platform + "/"
+    if locale:
+        url += locale + "/"
+    return url
+
+
+async def fetch_listing(session, url):
+    headers = {
+        "Accept": "application/json",
+        "User-Agent": "BuildHub;storage-team@mozilla.com"
+    }
+    async with session.get(url, headers=headers) as response:
+        try:
+            data = await response.json()
+            return data["prefixes"], data["files"]
+        except (aiohttp.ClientError, KeyError) as e:
+            raise ValueError("Could not fetch %s: %s" % (url, e))
+
+
+async def fetch_products(session, queue, products):
+    futures = [fetch_versions(session, queue, product) for product in products]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_versions(session, queue, product):
+    product_url = archive_url(product)
+    versions_folders, _ = await fetch_listing(session, product_url)
+
+    versions = [v[:-1] for v in versions_folders if re.match(r'^[0-9]', v)]
+    versions = [v for v in versions if "funnelcake" not in v]
+    versions = sorted(versions, key=lambda v: version_parse(v), reverse=True)
+
+    futures = [fetch_platforms(session, queue, product, version)
+               for version in versions]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_platforms(session, queue, product, version):
+    version_url = archive_url(product, version)
+    platform_folders, _ = await fetch_listing(session, version_url)
+
+    platforms = [p[:-1] for p in platform_folders]
+    platforms = [p for p in platforms if p not in ("source", "update")]
```

Comment:
I believe you put those on two separate lines to make it more readable 👍 
I would personally go one step further and comment the first line with `Strip the ending /`